xbox/xbox.c

```diff
@@ -0,0 +1,1002 @@
+        MASK(NV097_SET_COMBINER_CONTROL_FACTOR0, NV097_SET_COMBINER_CONTROL_FACTOR0_SAME_FACTOR_ALL)
+      | MASK(NV097_SET_COMBINER_CONTROL_FACTOR1, NV097_SET_COMBINER_CONTROL_FACTOR1_SAME_FACTOR_ALL)
+      | MASK(NV097_SET_COMBINER_CONTROL_ITERATION_COUNT, 4));
+  p = pb_push1(p, NV097_SET_COMBINER_SPECULAR_FOG_CW0,
+        MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW0_A_SOURCE, 0x0) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW0_A_ALPHA, 0) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW0_A_INVERSE, 0)
+      | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW0_B_SOURCE, 0x0) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW0_B_ALPHA, 0) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW0_B_INVERSE, 0)
+      | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW0_C_SOURCE, 0x0) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW0_C_ALPHA, 0) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW0_C_INVERSE, 0)
+      | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW0_D_SOURCE, 0x4) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW0_D_ALPHA, 0) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW0_D_INVERSE, 0));
+  p = pb_push1(p, NV097_SET_COMBINER_SPECULAR_FOG_CW1,
+        MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW1_E_SOURCE, 0x0) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW1_E_ALPHA, 0) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW1_E_INVERSE, 0)
+      | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW1_F_SOURCE, 0x0) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW1_F_ALPHA, 0) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW1_F_INVERSE, 0)
+      | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW1_G_SOURCE, 0x4) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW1_G_ALPHA, 1) | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW1_G_INVERSE, 0)
+      | MASK(NV097_SET_COMBINER_SPECULAR_FOG_CW1_SPECULAR_CLAMP, 0));
+
+  pb_end(p);
+}
+#undef MASK
+
+static GLuint gl_array_buffer = 0;
+static GLuint gl_element_array_buffer = 0;
+
+static GLuint* get_bound_buffer_store(GLenum target) {
+  switch(target) {
+  case GL_ARRAY_BUFFER:
+    return &gl_array_buffer;
+  case GL_ELEMENT_ARRAY_BUFFER:
+    return &gl_element_array_buffer;
+  default:
+    assert(false);
+    break;
+  }
+  return 0;  
+}
+
+
+static void* targetMatrix = NULL; //FIXME: Use this properly or remove it
+
+static void prepare_drawing() {
+  //FIXME: Track dirty bits
+
+  //FIXME: Set up all matrices etc.
+
+  // Set the register combiner
+  setup_texenv();
+}
+
+
+
+
+
+
+
+GL_API void GL_APIENTRY glGetIntegerv (GLenum pname, GLint *data) {
+  switch(pname) {
+  default:
+    assert(false);
+    break;
+  }
+}
+
+GL_API const GLubyte *GL_APIENTRY glGetString (GLenum name) {
+  switch(name) {
+  default:
+    assert(false);
+    break;
+  }
+  return (GLubyte*)"";
+}
+
+
+// Clearing
+GL_API void GL_APIENTRY glClear (GLbitfield mask) {
+
+  XguClearSurface flags = 0;
+  if (mask & GL_COLOR_BUFFER_BIT)   { flags |= XGU_CLEAR_COLOR;   }
+  if (mask & GL_DEPTH_BUFFER_BIT)   { flags |= XGU_CLEAR_Z;       }
+  if (mask & GL_STENCIL_BUFFER_BIT) { flags |= XGU_CLEAR_STENCIL; }
+
+  uint32_t* p = pb_begin();
+  //FIXME: Set clear region
```

Comment:
Reminder as this might crash

Reply:
No crash, but doesn't seem to work yet either. Did not try setting region yet - currently falling back to `pb_fill()`